### PR TITLE
drop volatile status dependency

### PR DESCRIPTION
**Pull request by mariusgrigoriu**
PR that might address #536 

This PR drops the dependency on the volatile status, which causes k8s_object and downstream dependencies to rebuild unnecessarily when using remote cache.

### CODE REVIEW

**Comment by mariusgrigoriu** [comment]
Apologies about the `+=` change. Seems that buildifier auto-formatter likes `.append` better for some reason. 

I mostly wanted to have a discussion about disabling volatile status stamping to work around https://github.com/bazelbuild/bazel/issues/10075 and whether this is an appropriate workaround or whether there is a better way.

**Comment by fejta** [comment]
Thanks for this, can you rebase?

/assign
/ok-to-test

> why change += to .append ?

Bazel will eventually require this

File: k8s/object.bzl [hunk 1]
```diff
@@ -89,17 +89,17 @@ def _impl(ctx):
...
-            image_specs += ["';'".join([
+            image_specs.append("';'".join([
                 "%s=%s" % (k, v)
                 for (k, v) in image_spec.items()
-            ])]
+            ]))
 
     # Add workspace_status_command files to the args that are pushed to the resolver and adds the
     # files to the runfiles so they are available to the resolver executable.
-    stamp_inputs = [ctx.info_file, ctx.version_file]
```
Comment (fejta) [inline k8s/object.bzl:102]: Strange, I cannot see how this is even used?

**Comment by mariusgrigoriu** [comment]
Looks like we would need to change https://github.com/bazelbuild/rules_k8s/blob/master/print-workspace-status.sh#L22 to `STABLE_E2E_NAMESPACE` and any other references to that stamp. Do you still need this rebased?

**Comment by fejta** [comment]
`E2E_NAMESPACE` is volatile -- it changes every build/PR -- why should this be changed to `STABLE_E2E_NAMESPACE`?

**Comment by mariusgrigoriu** [comment]
This PR attempts to work around the issue that volatile stamps trigger rebuilds when a remote cache is used by dropping the dependency on the volatile file. The CI job fails since volatile stamps are no longer available to the rule.

So is this really acceptable after all?

**Comment by fejta** [comment]
This repo isn't using a remote cache right now.

**Pull request closed without merging**